lib/elixir/lib/task/supervised.ex

```diff
@@ -189,20 +189,46 @@ defmodule Task.Supervised do
 
   ## Stream
 
-  def stream(enumerable, acc, reducer, callers, mfa, options, spawn) do
-    next = &Enumerable.reduce(enumerable, &1, fn x, acc -> {:suspend, [x | acc]} end)
-    max_concurrency = Keyword.get(options, :max_concurrency, System.schedulers_online())
+  def validate_stream_options!(options) do
+    options =
+      Keyword.validate!(options,
+        max_concurrency: System.schedulers_online(),
+        on_timeout: :exit,
+        timeout: 5000,
+        ordered: true,
+        zip_input_on_exit: false,
```

Comment:
Note: I didn't enforce these are booleans since truthiness works, not sure if we should be more strict at the risk of breaking code.